unit_scaling/functional.py

```diff
@@ -126,7 +192,7 @@ def dropout(
 def matmul(
     left: Tensor,
     right: Tensor,
-    constraint: Optional[str] = "gmean",
+    constraint: Optional[str] = "to_output_scale",
```

Comment:
Why the change to `to_output_scale` as default for matmul too?

Reply:
Good point.

I suppose if a client used `matmul` to implement something like `scaled_dot_product_attention` manually, they'd be more likely to get the right behaviour (consistent with the new unit-scaling principle) using `to_output_scale` than `gmean`. Not sure if that outweighs the "matmul is a general tool" thing, though.